sdk/core/core-util/src/checkEnvironment.ts

```diff
@@ -74,6 +74,11 @@ export const isNode =
   !isDeno &&
```

Comment:
GH did not let me comment over the right line (69) but I wonder if we should mark `isNode` as `@deprecated` so package owners know to use `isNodeCompatible` instead

Reply:
That's possible, @xirzec any thoughts?

Reply:
Feels a bit odd to me having `isDeno` and `isBun` around while deprecating `isNode`. I wonder if we will ever run into a scenario in future where we will be wanting to check for Node specifically, intentionally excluding Deno/Bun?

Reply:
Yeah I think we should deprecate the existing one if we're reverting the behavior to match isNodeCompatible (really just exporting it twice with different names.)

If we think there's a case for "is actually node" then perhaps we can use a worse name for it? like isNodeRuntime ? And then clarify in the doc comments all the options

Reply:
Oh if we're going to change the behavior of `isNode` to be the same as the new flag then I support deprecating but that doesn't seem to be what we're doing in this PR?